File: src/test/java/com/mysawit/shipment/MysawitShipmentServiceApplicationTests.java
package com.mysawit.shipment;

import org.junit.jupiter.api.Test;
import org.mockito.MockedStatic;
import org.springframework.boot.SpringApplication;

import static org.mockito.Mockito.*;

class MysawitShipmentServiceApplicationTests {

    @Test
    void constructorCanBeCalled() {
        new MysawitShipmentServiceApplication();
    }

    @Test
    void mainRunsSpringApplication() {
        String[] args = new String[] {"--spring.main.web-application-type=none"};
        try (MockedStatic<SpringApplication> springApplication = mockStatic(SpringApplication.class)) {
            MysawitShipmentServiceApplication.main(args);

            springApplication.verify(() -> SpringApplication.run(MysawitShipmentServiceApplication.class, args));
        }
    }
}

File: src/test/kotlin/com/mysawit/shipment/MysawitShipmentServiceApplicationTests.kt
package com.mysawit.shipment

import org.junit.jupiter.api.Test
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles

@SpringBootTest
@ActiveProfiles("test")
class MysawitShipmentServiceApplicationTests {

    @Test
    fun contextLoads() {
    }

    @Test
    fun mainRunsWithTestProfile() {
        main(
            arrayOf(
                "--spring.profiles.active=test",
                "--spring.main.web-application-type=none",
                "--app.test.close-context=true"
            )
        )
    }
}
